nixos/modules/services/system/userborn.nix

```diff
@@ -0,0 +1,174 @@
+{
+  utils,
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+
+  cfg = config.services.userborn;
+  userCfg = config.users;
+
+  userbornConfig = {
+    groups = lib.mapAttrsToList (username: opts: {
+      inherit (opts) name gid members;
+    }) config.users.groups;
+
+    users = lib.mapAttrsToList (username: opts: {
+      inherit (opts)
+        name
+        uid
+        group
+        description
+        home
+        password
+        hashedPassword
+        hashedPasswordFile
+        initialPassword
+        initialHashedPassword
+        ;
+      isNormal = opts.isNormalUser;
+      shell = utils.toShellPath opts.shell;
+    }) config.users.users;
+  };
+
+  userbornConfigJson = pkgs.writeText "userborn.json" (builtins.toJSON userbornConfig);
+
+  immutableEtc = config.system.etc.overlay.enable && !config.system.etc.overlay.mutable;
+  passwordFilesLocation = if immutableEtc then cfg.immutablePasswordFilesLocation else "/etc";
```

Comment:
I think that there may be other reasons to want to store these files outside of /etc, even if /etc is mutable. Like for instance when /etc is on a tmpfs.

Maybe we should just have an option that specifies the location to store the files at, that defaults to /etc, and we add an assertion to make sure that if /etc is immutable, the location is not /etc.

I think the current option is a bit too restrictive because the only way to have the files stored elsewhere is to also have etc.overlay enabled and set to immutable.